source/developers-guide/shopware-5-upgrade-guide-for-developers/index.md

```diff
@@ -222,6 +222,44 @@ If you want to implement this feature for your own element, you only need to pro
 interface `Shopware\Components\HttpCache\CacheTimeServiceInterface` and then tag this service in the `services.xml` with
 the tag `invalidation_date_provider`. It will then be picked up and called automatically.
 
+### Cache warmer
+
+Up to know, the cache warmer relied on seo urls, but that did only cover a small amount of possible urls which can be found
+in shopware. With these changes, developers doesn't have to add unnecessary new seo urls to warm them for the cache.
+In addition, the performance and amount of urls were greatly improved to cover the most content of shopware by itself.
+
+By implementing `HttpCache\UrlProvider\UrlProviderInterface` to a new service with the tag `cache_warmer.url_provider`
+developers can now easily add their own url providers for the cache warmer and also call these providers independently.
+These will now be called by the CLI module after updating `prepareOptions`. You can optionally add a parameter for it to the
```

Comment:
so that means if I want to call my own URL provider, I have to change some code in the core? because with a plugin you cannot change a command